importlib_metadata/__init__.py

```diff
@@ -494,12 +494,19 @@ def __init__(self, name):
         self.name = name
         if name is None:
             return
-        self.normalized = name.lower().replace('-', '_')
+        self.normalized = self.normalize(name)
         self.prefix = self.normalized + '-'
         self.exact_matches = [
             self.normalized + suffix for suffix in self.suffixes]
         self.versionless_egg_name = self.normalized + '.egg'
 
+    @staticmethod
+    def normalize(name):
+        """
+        PEP 503 normalization plus dashes as underscores.
+        """
+        return re.sub(r"[-_.]+", "-", name).lower().replace('-', '_')
```

Comment:
Can this not be simplified to `re.sub(r"[-_.]+", "_", name).lower()`?

Reply:
Yes, it could. But then it becomes less clear (modifies instead of extends PEP 503). I'm divided as to which approach is more readable. Perhaps the `dist-info` docs should provide a reference implementation and this library could adopt that implementation verbatim.

Reply:
Please commit whatever your preference is here. I'm happy to have something that's compatible in place and iterate more later if needed.